scheduler/src/cook/kubernetes/api.clj

```diff
@@ -184,6 +184,39 @@
            (-> m (get "cpu") to-double)
            0.0)})
 
+(defn get-taint-key
+  "Get the taint variable out of a taint key out of the V1Taint object."
+  [^V1Taint taint]
+  (.getKey taint))
+
+(defn process-taints
+  "Given a map from node-name to node, generate a map from node-name->processed taints.
+  If there is no taint of the form cook.pools, return 'no-pool'. Processing a taints involves
+  splitting the list into the cook.pool taint and the other taints."
+  [node-name->node]
+  (pc/map-vals (fn [^V1Node node]
+                 (let [taints-on-node (or (some-> node .getSpec .getTaints) [])]
+                   (group-by #(if (= "cook.pool" (get-taint-key %)) :pool-taint :other-taint)  taints-on-node))) node-name->node))
+
+(defn get-node-pool
+  "Get the pool for a node. In the case of no pool, return 'no-pool"
+  [node->processed-taints node-name]
```

Comment:
What about changing this to just take a Node?

Reply:
Done.